src/common/types/int128_t.cpp

```diff
@@ -0,0 +1,633 @@
+    result = value;
+    return true;
+}
+
+template<>
+bool Int128_t::tryCastTo(float value, int128_t& result) {
+    return tryCastTo(double(value), result);
+}
+
+template<class REAL_T>
+bool castFloatingToInt128(REAL_T value, int128_t& result) {
+    // TODO: Maybe need to add func isFinite in value.h to see if every type is finite.
+    if (value <= -170141183460469231731687303715884105728.0 ||
+        value >= 170141183460469231731687303715884105727.0) {
+        return false;
+    }
+    bool negative = value < 0;
+    if (negative) {
+        value = -value;
+    }
+    result.low = (uint64_t)fmod(value, REAL_T(std::numeric_limits<uint64_t>::max()));
+    result.high = (uint64_t)(value / REAL_T(std::numeric_limits<uint64_t>::max()));
+    if (negative) {
+        Int128_t::negateInPlace(result);
+    }
+    return true;
+}
+
+template<>
+bool Int128_t::tryCastTo(double value, int128_t& result) {
+    return castFloatingToInt128<double>(value, result);
+}
+
+template<>
+bool Int128_t::tryCastTo(long double value, int128_t& result) {
+    return castFloatingToInt128<long double>(value, result);
+}
+
+//===============================================================================================
+
+int128_t::int128_t(int64_t value) {
+    auto result = Int128_t::castTo(value);
+    this->low = result.low;
+    this->high = result.high;
+}
+
+//===============================================================================================
+bool int128_t::operator==(const int128_t& rhs) const {
+    return Int128_t::Equals(*this, rhs);
+}
+
+bool int128_t::operator!=(const int128_t& rhs) const {
+    return Int128_t::notEquals(*this, rhs);
+}
+
+bool int128_t::operator>(const int128_t& rhs) const {
+    return Int128_t::greaterThan(*this, rhs);
+}
+
+bool int128_t::operator>=(const int128_t& rhs) const {
+    return Int128_t::greaterThanOrEquals(*this, rhs);
+}
+
+bool int128_t::operator<(const int128_t& rhs) const {
+    return Int128_t::lessThan(*this, rhs);
+}
+
+bool int128_t::operator<=(const int128_t& rhs) const {
+    return Int128_t::lessThanOrEquals(*this, rhs);
+}
+
+int128_t int128_t::operator+(const int128_t& rhs) const {
+    return Int128_t::Add(*this, rhs);
+}
+
+int128_t int128_t::operator-(const int128_t& rhs) const {
+    return Int128_t::Sub(*this, rhs);
+}
+
+int128_t int128_t::operator*(const int128_t& rhs) const {
+    return Int128_t::Mul(*this, rhs);
+}
+
+int128_t int128_t::operator/(const int128_t& rhs) const {
+    return Int128_t::Div(*this, rhs);
+}
+
+int128_t int128_t::operator%(const int128_t& rhs) const {
+    return Int128_t::Mod(*this, rhs);
+}
+
+int128_t int128_t::operator-() const {
+    return Int128_t::negate(*this);
+}
+
+// inplace arithmetic operators
+int128_t& int128_t::operator+=(const int128_t& rhs) {
+    if (!Int128_t::addInPlace(*this, rhs)) {
+        throw common::OverflowException("INT128 is out of range: cannot add in place.");
```

Comment:
Use codecov instructions to ignore the coverage check for this exception.
BTW: When do we need to call inplace add?

Reply:
in list_sum_function case for int128